lib/bollinger.js

```diff
@@ -1,14 +1,26 @@
 // Bollinger Bands
 var bollingerbands = require('bollinger-bands')
+
 module.exports = function bollinger (s, key, length, source_key) {
   if (!source_key) source_key = 'close'
   if (s.lookback.length > length) {
+    // skip calculation if result already presented as we use historical data only,
+    // no need to recalculate for each individual trade
+    if (key in s.period) return
```

Comment:
This is performance improvement. As I found we call this for each individual trade, but if bands calculated we do not need to recalculate them